sirepo/package_data/static/js/sirepo.js

```diff
@@ -1907,13 +1963,35 @@ SIREPO.app.controller('SimulationsController', function (appState, panelState, r
     }
 
     function setActiveFolder(item) {
+        var prevPath = self.pathName(self.activeFolder || item);
         self.activeFolder = item;
         self.activeFolderPath = [];
         while (item) {
             self.activeFolderPath.unshift(item);
             item = item.parent;
         }
         appState.setActiveFolderPath(self.pathName(self.activeFolder));
+        appState.setActiveFolder(self.activeFolder);
+        if( prevPath !== self.pathName(self.activeFolder) ) {
+            var compoundPath = pathToCompoundPath(self.pathName(self.activeFolder));
+            if( compoundPath == '' ) {
+                requestSender.localRedirect('simulations');
+            }
+            else {
+                var params = {
+                    ':folderPath?': compoundPath,
+                };
+                requestSender.localRedirect('simulationsFolder', params);
+            }
+        }
+    }
+
+    function pathToCompoundPath(path) {
+        return path.replace(/^\//,'').replace(/\//g, '_');
```

Comment:
I don't understand this replacement. '_' is a legitimate path character so it can't be used as a separator. These two paths are equal as compound paths: '/a_b' and '/a/b'. However, that's incorrect. You have to use a character that is not a valid path character as a separator. Not sure why compound path is needed at all.

Reply:
Right, I meant to ask what to use.  I would advocate for something that does not get url encoded...how about "!" like in old-school email addresses?  Then just add it to the list of forbidden characters

Reply:
Let's not add more invalid chars. We already have a good enough list, e.g. could use ':'. 

I wonder about compoundpath, however. why is it necessary?

Reply:
I think angular can't find the route if we use "/" -- but I may be misinterpreting your question. BTW I believe ":" will fail for a similar reason (it will be interpreted as a route param) but I'll try it

Reply:
Followup - colon is fine as it gets url encoded. I find the resulting URLs hard to parse visually, but barring user feedback to that effect it doesn't matter I suppose

Reply:
Given the way angular works, we might be better off using a search (query) parameter:

```text
/simulations?p=/a/b/c
```

This will produce a readable URL. If angular had the concept of path info, we'd be fine, but it doesn't. Using "search" has the added benefit of not requiring two routes as empty searches work just fine.